File: openapi-generator/csharp-dotnet2/src/main/CsharpDotNet2/Org/OpenAPITools/Model/AssetReeferResponseReeferStatsPowerStatus.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace Org.OpenAPITools.Model {

  /// <summary>
  /// 
  /// </summary>
  [DataContract]
  public class AssetReeferResponseReeferStatsPowerStatus {
    /// <summary>
    /// Timestamp in Unix milliseconds since epoch.
    /// </summary>
    /// <value>Timestamp in Unix milliseconds since epoch.</value>
    [DataMember(Name="changedAtMs", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "changedAtMs")]
    public long? ChangedAtMs { get; set; }

    /// <summary>
    /// Power status of the reefer.
    /// </summary>
    /// <value>Power status of the reefer.</value>
    [DataMember(Name="status", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "status")]
    public string Status { get; set; }


    /// <summary>
    /// Get the string presentation of the object
    /// </summary>
    /// <returns>String presentation of the object</returns>
    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class AssetReeferResponseReeferStatsPowerStatus {\n");
      sb.Append("  ChangedAtMs: ").Append(ChangedAtMs).Append("\n");
      sb.Append("  Status: ").Append(Status).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }

    /// <summary>
    /// Get the JSON string presentation of the object
    /// </summary>
    /// <returns>JSON string presentation of the object</returns>
    public string ToJson() {
      return JsonConvert.SerializeObject(this, Formatting.Indented);
    }

}
}

File: openapi-generator/dart/lib/model/asset_reefer_response_reefer_stats_power_status.dart
part of openapi.api;

class AssetReeferResponseReeferStatsPowerStatus {
  /* Timestamp in Unix milliseconds since epoch. */
  int changedAtMs = null;
  /* Power status of the reefer. */
  String status = null;
  AssetReeferResponseReeferStatsPowerStatus();

  @override
  String toString() {
    return 'AssetReeferResponseReeferStatsPowerStatus[changedAtMs=$changedAtMs, status=$status, ]';
  }

  AssetReeferResponseReeferStatsPowerStatus.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    if (json['changedAtMs'] == null) {
      changedAtMs = null;
    } else {
          changedAtMs = json['changedAtMs'];
    }
    if (json['status'] == null) {
      status = null;
    } else {
          status = json['status'];
    }
  }

  Map<String, dynamic> toJson() {
    return {
      'changedAtMs': changedAtMs,
      'status': status
    };
  }

  static List<AssetReeferResponseReeferStatsPowerStatus> listFromJson(List<dynamic> json) {
    return json == null ? new List<AssetReeferResponseReeferStatsPowerStatus>() : json.map((value) => new AssetReeferResponseReeferStatsPowerStatus.fromJson(value)).toList();
  }

  static Map<String, AssetReeferResponseReeferStatsPowerStatus> mapFromJson(Map<String, dynamic> json) {
    var map = new Map<String, AssetReeferResponseReeferStatsPowerStatus>();
    if (json != null && json.length > 0) {
      json.forEach((String key, dynamic value) => map[key] = new AssetReeferResponseReeferStatsPowerStatus.fromJson(value));
    }
    return map;
  }
}

